pkg/services/preference/prefimpl/pref.go

```diff
@@ -2,35 +2,49 @@ package prefimpl
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
+	"fmt"
 	"time"
 
 	"github.com/grafana/grafana/pkg/infra/db"
+	"github.com/grafana/grafana/pkg/kinds/preferences"
+	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/featuremgmt"
 	pref "github.com/grafana/grafana/pkg/services/preference"
+	"github.com/grafana/grafana/pkg/services/sqlstore/session"
+	"github.com/grafana/grafana/pkg/services/store/entity"
 	"github.com/grafana/grafana/pkg/setting"
 )
 
 type Service struct {
-	store    store
-	cfg      *setting.Cfg
-	features *featuremgmt.FeatureManager
+	store       store
+	cfg         *setting.Cfg
+	features    *featuremgmt.FeatureManager
+	entitystore entity.EntityStoreServer
+	sess        *session.SessionDB
```

Comment:
This makes me very nervous but I'm not yet sure if that's based on actual concerns (my db experience is mostly provisioning infra) - I'm going to get a second opinion from the backend squad.

Reply:
If you mean creating a session at startup and then using it everywhere... i agree!  but that seems to be the pattern used everywhere.  I *think* the `session.SessionDB` actually handles refreshing sessions as needed -- if it did not, everything would be broken

Reply:
I'm sorry, my most recent comments here were typed dec 1st and never posted 🤦